Y/Yosys/build_tarballs.jl

```diff
@@ -3,60 +3,65 @@
 using BinaryBuilder, Pkg
 
 name = "Yosys"
-version = v"0.11.0"
+version = v"0.18.0"
 
 # Collection of sources required to complete build
 sources = [
-    GitSource("https://github.com/YosysHQ/yosys.git", "360fed8e4d611fa725a4526cf960383b8e6c6e64")
+    GitSource("https://github.com/YosysHQ/yosys.git", "19ce3b45d6a877e96af3883e3f8df3acfd224ea6")
 ]
 
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = filter!(p -> arch(p) != "armv6l", supported_platforms())
+platforms = expand_cxxstring_abis(platforms)
+# For some reasons, building for CXX03 string ABI doesn't actually work, skip it
+filter!(x -> cxxstring_abi(x) != "cxx03", platforms)
+
 dependencies = [
     Dependency("boost_jll"; compat="=1.76.0"), # max gcc7
     Dependency("Readline_jll"; compat="8.1.1"),
     Dependency("Tcl_jll"; compat="8.6.11"),
     Dependency("Zlib_jll"; compat="1.2.11"),
```

Comment:
This is too restrictive
```suggestion
    Dependency("Zlib_jll"; compat="1.2.12"),
```
(and it doesn't even seem to be needed, since 1.2.11 isn't available on aarch64-darwin: https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=20106&view=logs&j=e67459c4-ac44-5ab7-7440-351b23770d00&t=223112fb-788d-5803-3e8e-8d0b704125af&l=77)

Reply:
Ugh, I thought this had been fixed before merging :expressionless: